### PR TITLE
Display copy icon and adding send action to the link menu

### LINKED ISSUE #2705: How to copy share link
### Actual behaviour
- The new files details / sharing screen looks great but i can´t find a way to copy the link. Am i doing sth. wrong?

### Expected behaviour
- The context menu should show sth. like "copy link"
 
### Steps to reproduce
1. Share a file from the web browser via link
2. Try to share the url from within the Nc App

### Environment data
Android version: Android 8.1

Device model: OnePlus 6

Stock or customized system: Stock

Nextcloud app version: 3.2.0

Nextcloud server version: 13.0.4

### Screenshots

From the Nc Blog post:

share

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Fix #2705 

device-2018-06-19-165059

device-2018-06-19-165115

Beware that I changed the behavior a bit as-in copy to clipboard is now a Toast, not a Snackbar. I checked the behavior for the Android OS and typical (Google) apps and they all show a Toast for this.

@jancborchardt I did change the layout a bit (see screenshots) since aligning the copy icon next to the 3-dot menu seemed strange since shared folders also have the "edit option" for link-share so I thought it be better to add a small copy-icon-badge to the text to indicate  it can be clicked and moved the "send link" functionality to the link's menu (see screenshot). Is this fine with you?

Also @tobiasKaminsky @mario please review :heart: 

Furthermore I think this should be put into 3.2.1 so I vote to have a backport :tada:

### CODE REVIEW

**Comment by mario** [comment]
👍 
Minor codacy stuff.

**Comment by AndyScherzinger** [comment]
Thanks @mario :heart: Just pushed the fix for codacy :)

File: src/main/java/com/owncloud/android/utils/ClipboardUtil.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                ClipData clip = ClipData.newPlainText(
+                        activity.getString(
+                                R.string.clipboard_label, activity.getString(R.string.app_name)),
+                        text
+                );
+                ((ClipboardManager) activity.getSystemService(Context.CLIPBOARD_SERVICE)).setPrimaryClip(clip);
+
+                Toast.makeText(activity, R.string.clipboard_text_copied, Toast.LENGTH_SHORT).show();
+            } catch (Exception e) {
+                Toast.makeText(activity, R.string.clipboard_uxexpected_error, Toast.LENGTH_SHORT).show();
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ClipboardUtil.java]: Please fix the typo, though it is not by you ;-)

**Comment by tobiasKaminsky** [comment]
:+1: 
If "share link" is clickable, the copy icon is, for me, a bit too narrow to it.

**Comment by AndyScherzinger** [comment]
Will fix typo and whitespace later today :)

Thank you all for the feedback and review 👍

**Comment by jancborchardt** [comment]
Looks good! I would also only have the feedback of the copy icon needing more whitespace around :)

**Comment by jancborchardt** [comment]
Looks quite similar to the first screenshot it seems. The icon needs to be moved a bit down (verticaly centered with the checkboxes and text) and a bit more to the right.

**Comment by AndyScherzinger** [comment]
Ah, got it. I always tried to have it as some kind of a badge while you folks want it to me bore like a menu icon (size, whitespace, etc.):
device-2018-06-20-114432

**Comment by jancborchardt** [comment]
Yesss, perfect! :)

**Pull request merged**